source_gen/lib/src/generator_for_annotation.dart

```diff
@@ -93,5 +99,28 @@ abstract class GeneratorForAnnotation<T> extends Generator {
     Element element,
     ConstantReader annotation,
     BuildStep buildStep,
-  );
+  ) {}
+
+  /// Implement to return source code to generate for [element].
+  ///
+  /// This method is invoked based on finding elements annotated with an
+  /// instance of [T]. The [annotation] is provided as a [ConstantReader].
+  ///
+  /// Supported return values include a single [String] or multiple [String]
+  /// instances within an [Iterable] or [Stream]. It is also valid to return a
+  /// [Future] of [String], [Iterable], or [Stream]. When multiple values are
+  /// returned through an iterable or stream they will be deduplicated.
+  /// Typically each value will be an independent unit of code and the
+  /// deduplication prevents re-defining the same member multiple times. For
+  /// example if multiple annotated elements may need a specific utility method
+  /// available it can be output for each one, and the single deduplicated
+  /// definition can be shared.
+  ///
+  /// Implementations should return `null` when no content is generated. Empty
+  /// or whitespace-only [String] instances are also ignored.
+  dynamic generateForAnnotatedElement2(
```

Comment:
Wait, so which version is going to be called?

Reply:
Both, but any specific generator will override only one.
So, both migrated and not-yet-migrated generators will work.